.github/workflows/updatecli.yml

```diff
@@ -0,0 +1,42 @@
+name: "Updatecli: Dependency Management"
+
+on:
+  schedule:
+    # Run every hour
+    - cron: '0 * * * *'
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+permissions:
+  contents: write
+  issues: write
+  pull-requests: write
+
+jobs:
+  updatecli:
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: '1.19.3'
```

Comment:
Does this align with the Golang version used in the repository/project that it is being used for?

Reply:
It appears that Golang here is used for rancher/helm wich use Golang 1.17.
I think it would be usefull to comment why Golang is used for and when to update to a different version

Reply:
Go is used during the update of rancher/shell in Prometheus Federator in https://github.com/rancherlabs/updatecli-automation/blob/e9be9ace812b529141cfbaffcffbe3df3aed8800/updatecli/scripts/prometheus-federator-update-commit-hash.sh#L14-L15 and https://github.com/rancherlabs/updatecli-automation/blob/e9be9ace812b529141cfbaffcffbe3df3aed8800/updatecli/updatecli.d/rancher/shell/prometheus-federator/shell.yaml#L121.

I'll add a comment in those lines in the GH workflow to explain their need.

Reply:
Updated.